src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs

```diff
@@ -3929,41 +3920,37 @@ ParameterPeekAheadValue peekAhead
                     enumerator = value.GetEnumerator();
                 }
 
-                using (enumerator)
```

Comment:
The using statement here lowers in the compiler to a try finally block with disposal in the finally block. This means that this code will try to dispose the `enumerator` object twice. Doing this might be harmless but  it will depend on the specific implementation of the enumerator, it should really throw an ObjectDisposedException.